RhetosPackages/Source/Angular2ModelGenerator/Angular2ModelGenerator.cs

```diff
@@ -69,18 +72,36 @@ public void Generate()
 
         private void CompileFileTS()
         {
+            var arguments = $"/C cd /d {Paths.ResourcesFolder}/AdminGuiCompile/scripts & npm install --loglevel=error & npm run tsc";
             var process = new Process
             {
                 StartInfo = new ProcessStartInfo
                 {
-                    WindowStyle = ProcessWindowStyle.Hidden,
                     FileName = "cmd.exe",
-                    Arguments = "/C cd /d " + Paths.ResourcesFolder + "/AdminGuiCompile/scripts" + " & tsc",
+                    Arguments = arguments,
+                    CreateNoWindow = true,
+                    UseShellExecute = false,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true
                 }
             };
 
+            _deployPackagesLogger.Trace("Compiling TypeScript model file.");
+            
             process.Start();
-            process.WaitForExit();
+            try
+            {
+                _deployPackagesLogger.Trace(process.StandardOutput);
+                if (process.StandardError.Peek() > -1)
+                {
+                    _deployPackagesLogger.Error(process.StandardError);
+                    throw new TypeScriptException("Compiled TypeScript model file failed.");
```

Comment:
Message should be "Compiling TypeScript models failed", or "Failed to compile TypeScript models."

Reply:
Agree :+1: